sdk/samples/iot/hub/src/paho_iot_hub_pnp_example.c

```diff
@@ -0,0 +1,1020 @@
+// temp reported prop.
+static void handle_twin_message(
+    MQTTClient_message* message,
+    az_iot_hub_client_twin_response* twin_response)
+{
+  az_result result;
+  uint32_t desired_temp;
+  int32_t  version_num;
+  az_span twin_payload_span
+      = az_span_init((uint8_t*)message->payload, (int32_t)message->payloadlen);
+  // Determine what type of incoming twin message this is. Print relevant data for the message.
+  switch (twin_response->response_type)
+  {
+    // A response from a twin GET publish message with the twin document as a payload.
+    case AZ_IOT_CLIENT_TWIN_RESPONSE_TYPE_GET:
+      printf("A twin GET response was received\n");
+      if (az_failed(result =
+              parse_twin_desired_temperature_property(twin_payload_span, true, &desired_temp, &version_num)))
+      {
+        // If the item can't be found, the desired temp might not be set so take no action
+        break;
+      }
+      else
+      {
+        send_reported_temperature_property(desired_temp, version_num);
+
+        if (update_device_temp((int32_t)desired_temp))
+        {
+          send_reported_temperature_property(device_max_temp, version_num);
+        }
+      }
+      if (message->payloadlen)
+      {
+        printf("Payload:\n%.*s\n", message->payloadlen, (char*)message->payload);
+      }
+      break;
+    // An update to the desired properties with the properties as a JSON payload.
+    case AZ_IOT_CLIENT_TWIN_RESPONSE_TYPE_DESIRED_PROPERTIES:
+      printf("A twin desired properties message was received\n");
+      printf("Payload:\n%.*s\n", message->payloadlen, (char*)message->payload);
+
+      // Get the new temperature
+      if (az_failed(
+              result = parse_twin_desired_temperature_property(
+                  twin_payload_span, false, &desired_temp, &version_num)))
+      {
+        printf("Could not parse desired temperature property, az_result %04x\n", result);
+        break;
+      }
+      send_reported_temperature_property(desired_temp, version_num);
+
+      if (update_device_temp((int32_t)desired_temp))
+      {
+        send_reported_temperature_property(device_max_temp, -1);
+      }
+      break;
+
+    // A response from a twin reported properties publish message. With a successfull update of
+    // the reported properties, the payload will be empty and the status will be 204.
+    case AZ_IOT_CLIENT_TWIN_RESPONSE_TYPE_REPORTED_PROPERTIES:
+      printf("A twin reported properties response message was received\n");
+      break;
+  }
+}
+
+// Invoke the requested command if supported and return status | Return error otherwise
+static void handle_command_message(
+    MQTTClient_message* message,
+    az_iot_hub_client_method_request* command_request)
+{
+
+  if (az_span_is_content_equal(report_command_name_span, command_request->name))
+  {
+    az_span command_response_span = AZ_SPAN_FROM_BUFFER(commands_response_payload);
+    az_span command_payload_span
```

Comment:
>command_payload_span [](start = 12, length = 20)

Nit: Consider making this command_request_span, so that its intent is clearer.  "payload" in theory could be a request or response.